backend/src/main/java/com/github/NicolaiKopka/web_page_services/user_pages/user_favorites/UserFavoritesService.java

```diff
@@ -32,20 +32,24 @@ public Collection<Movie> getAllFavoriteMoviesFromDbByUser(String username) {
 
         return userFavorites.getMovieIds().stream().map(movieDBApiConnect::getMovieById).toList();
     }
-    public UserFavoritesSaveObject addMovieToFavorites(Integer movieId, String username) {
+    public void addMovieToFavorites(Integer movieId, String username) {
 
         MyUser user = myUserRepo.findByUsername(username).orElseThrow();
 
-        try {
-            UserFavoritesSaveObject userFavorites = userFavoritesRepo.findByUserId(user.getId()).orElseThrow();
-            userFavorites.addMovieId(movieId);
-            return userFavoritesRepo.save(userFavorites);
-        } catch (NoSuchElementException e) {
-            UserFavoritesSaveObject userFavorites = new UserFavoritesSaveObject();
-            userFavorites.setUserId(user.getId());
-            userFavorites.addMovieId(movieId);
-            return userFavoritesRepo.save(userFavorites);
-        }
+        userFavoritesRepo.findByUserId(user.getId()).ifPresentOrElse(
+                userFavorites -> {
+                    userFavorites.addMovieId(movieId);
+                    userFavoritesRepo.save(userFavorites);
+                },
+                () -> {
+                    UserFavoritesSaveObject userFavorites = new UserFavoritesSaveObject();
+                    userFavorites.setUserId(user.getId());
+                    userFavorites.addMovieId(movieId);
+                    userFavoritesRepo.save(userFavorites);
+                }
+        );
+
+        //return userFavoritesRepo.findByUserId(user.getId()).orElseThrow();
```

Comment:
You only need to save the object at one point. Save userFavoritesRepo.findByUserId() manipulate it afterwards, save to db and return it. Or maybe even better: just return userFavoritesRepo.save. It should also return the manipulated object